src/kibana/components/doc_table/lib/get_sort.js

```diff
@@ -5,38 +5,56 @@ define(function (require) {
    * Take a sorting array and make it into an object
    * @param {array} 2 item array [fieldToSort, directionToSort]
    * @param {object} indexPattern used for determining default sort
+   * @param {boolean} asDefaultSort always return an object as { fieldName: direction }
    * @returns {object} a sort object suitable for returning to elasticsearch
    */
-  function getSort(sort, indexPattern) {
+  function getSort(sort, indexPattern, asDefaultSort) {
     var sortObj = {};
     var field, direction;
 
-    function isSortable(field) {
-      return (indexPattern.fields.byName[field] && indexPattern.fields.byName[field].sortable);
+    function useTimeField() {
+      var timeField = (indexPattern.timeFieldName && indexPattern.fields.byName[indexPattern.timeFieldName]);
+      if (timeField && timeField.sortable) {
+        field = timeField;
+        direction = 'desc';
+      } else {
+        field = direction = undefined;
+      }
     }
 
-    if (_.isArray(sort) && sort.length === 2 && isSortable(sort[0])) {
-      // At some point we need to refact the sorting logic, this array sucks.
-      field = sort[0];
-      direction = sort[1];
-    } else if (indexPattern.timeFieldName && isSortable(indexPattern.timeFieldName)) {
-      field = indexPattern.timeFieldName;
-      direction = 'desc';
+    if (_.isArray(sort) && sort.length === 2) {
+      field = indexPattern.fields.byName[sort[0]];
+
+      if (field && field.sortable) {
+        // At some point we need to refact the sorting logic, this array sucks.
+        direction = sort[1];
+      } else {
+        useTimeField();
+      }
+    } else {
+      useTimeField();
     }
 
     if (field) {
-      sortObj[field] = direction;
+      // sorting on a scripted field requires the script value
+      if (field.scripted && !asDefaultSort) {
+        sortObj._script = {
```

Comment:
I think that a better place for this is [in the courier](https://github.com/elastic/kibana/blob/master/src/kibana/components/courier/data_source/_normalize_sort_request.js)

In order to accomplish everything there you will need the indexPattern, but that can be passed in from [the search source](https://github.com/elastic/kibana/blob/master/src/kibana/components/courier/data_source/search_source.js#L182) by calling `this.get('index')`